reflection_refraction.py

```diff
@@ -0,0 +1,184 @@
+    print("By Martin Lambov")
+    print("GitHub: https://github.com/5neepy")
+
+def check_alfa(alfa: float) -> None:
+    '''Check if the equation is possible by giving alfa certain range.
+
+    :param alfa: Angle of incidence.
+    '''
+    
+    if alfa >= 90 or alfa <= 0:
+        print("\nAlfa can't be equal or more than 90 and can't be equal or less than 0.\n")
+        exit()
+
+def check_index_of_refr(n1: float, n2: float) -> None:
+    '''Check if the refraction indices are equal.
+    
+    :param n1: Index of refraction of the first medium
+    :param n2: Index of refraction of the second medium
+    '''
+    
+    if n1 == n2:
+        print("\nIf n_1 = n_2 we don't have Reflection/Refraction of light.\n")
+        exit()
+
+def get_vars() -> tuple[float,float,float]:
+    '''Getting the variables from the user.
+    
+    :return: Angle of incidence.
+    :return: Index of refraction of the first medium
+    :return: Index of refraction of the second medium
+    '''
+    
+    alfa = float(input("\nEnter the degrees of alfa: "))
+    check_alfa(alfa)
+
+    print("\nEnter the indexes of refraction of usual objects: \n air = 1 \n water = 1.33 \n glass = 1.5")
+
+    n1 = float(input("\nn1: "))
+    n2 = float(input("n2: "))
+
+    check_index_of_refr(n1, n2)
+
+    return alfa, n1, n2
+
+def calculate_refr_angl(alfa: float, n1: float, n2: float) -> tuple[float,float,float]:
+    '''Calculate the refraction angle.
+    
+    :param alfa: Angle of incidence
+    :param n1: Index of refraction of the first medium
+    :param n2: Index of refraction of the second medium
+    :return: Angle of incidence
+    :return: alfa_prim: Angle of reflection
+    :return: beta: Angle of refraction
+    '''
+    
+    alfa_prim = alfa
+    
+    tot_intern_refl = lambda theta, n1, n2: theta >= 45 and n1 > n2
+    
+    if tot_intern_refl(alfa, n1, n2):
+        beta = 90
+    else:
+        try:
+            beta = math.degrees(math.asin(n1*math.sin(math.radians(alfa))/n2))  # n1*sin(θ1) = n2*sin(θ2)
+        except ValueError:
+            beta = None
+
+    return alfa, alfa_prim, beta
+ 
+def calc_ref_plot(alfa: float, beta:float) -> tuple[float,float,float]:
+    '''Calculate the positions of the points in the plot.
+    :param alfa: Angle of incidence
+    :param: beta: Angle of refraction
+    :return: Start cordinates of the incident beam
+    :return: Start cordinates of the reflected beam
+    :return: Start cordinates of the refracted beam
+    '''
+    
+    # Incident beam = a:
+    alfa_2 = 90+alfa
+    X_1 = (0+DISTANCE*math.cos(math.radians(alfa_2)))
+    Y_1 = (0+DISTANCE*math.sin(math.radians(alfa_2)))
+
+    # Reflected beam = a_1:
+    X_2 = -X_1
+    Y_2 = Y_1
+
+    if beta is not None:
+        # Refracted beam = b:
+        beta_2 = 270+beta
+        X_3 = (0+DISTANCE*math.cos(math.radians(beta_2)))
+        Y_3 = (0+DISTANCE*math.sin(math.radians(beta_2)))
+        p3 = [X_3, Y_3]
+    else:
+        p3 = None
```

Comment:
You can leave it as it is if you prefer. 

I think that it will also work if you write 
```
    if beta is not None:
        p3 = [X_3, Y_3]
    else:
        p3 = [0,0]
```

or equivalently you can write:
```
p3 = [X_3, Y_3] if beta is not None else [0,0]
``` 

then you won't need the `if p3 is not None:` below.